### ISSUE TITLE
AWS EC2 privacy and security

### ISSUE THREAD

**Issue by roishillo** [4dn-dcic/tibanna#275]
Hi Soo Lee,
We are going to use Tibanna with a great scenario for one of the Israeli Universities.
We would like to run Tibanna on AWS, and for that, we would love to develop some features and we need to understand whats the current status and if you can help out with the design.

Two practical questions - 
1. Is there a way to add the new EC2 to a specific security group, VPC etc.?
2. How the scale-up and down works for the EC2? how do we define them?

Do you think we can jump to a call and discuss it further?

Thanks,
Roi Shillo

**Comment by SooLee** [comment]
Hi Roi Shillo,

You can specify security group / subnet by adding 'security_group' : <security group id> and 'subnet' : <subnet id> in the config (you can use one or both of these to specify VPC. If you don't have default VPC, you'd have to specify subnet.). It's not in the documentation yet. I will try to update it soon.

Tibanna doesn't require scaling up and down. It launches an independent EC2 instance for each job with its own instance type / EBS rather than using an autoscaling feature. To use it in a large scale project, it may be helpful to request AWS for EC2 limit increase to be able to run more instances simultaneously.

**Comment by roishillo** [comment]
Thank you Soo for your quick reply.

Do you think we can have a short call to discuss the opportunity?
Can we talked in Linkedin (Roi Shillo) and setup a short session?

We would love to learn how to be a contributor for this open source project.

Thanks,
Roi Shillo

**Comment by freezecoder** [comment]
Hi All,

I encountered this exact issue when trying to set up Tibanna for the first time.  Before trying to supply a subnet all my 'hello world' workflows were failing and I traced  it back to it unable to start an instance due to requirement for VPC info. This only happens in aws with newer instance types like c4,m5, which is probably a default for Tibanna. When I set the instance type to something like "m1.large" 

```
"instance_type": "m1.large",

```

It started working and tibanna was running the workflow as expected.

Just for  kicks I tried the suggestion above and I entered an existing subnet ID created from aws VPC and then it worked without needing to change the instance type.
